src/Pkg.jl

```diff
@@ -80,18 +82,34 @@ Used as an argument to  [`PackageSpec`](@ref) or as an argument to [`Pkg.update`
 """
 const UpgradeLevel = Types.UpgradeLevel
 
+const PreserveLevel = Types.PreserveLevel
+
 # Define new variables so tab comleting Pkg. works.
 """
-    Pkg.add(pkg::Union{String, Vector{String}})
-    Pkg.add(pkg::Union{PackageSpec, Vector{PackageSpec}})
+    Pkg.add(pkg::Union{String, Vector{String}}; preserve=PRESERVE_TIERED)
+    Pkg.add(pkg::Union{PackageSpec, Vector{PackageSpec}}; preserve=PRESERVE_TIERED)
 
 Add a package to the current project. This package will be available by using the
 `import` and `using` keywords in the Julia REPL, and if the current project is
 a package, also inside that package.
 
+## Resolution Tiers
+`Pkg` resolves the set of packages in your environment using a tiered algorithm.
+The `preserve` keyword argument allows you to key into a specific tier in the resolve algorithm.
+The following table describes the argument values for `preserve` (in order of strictness):
+
+| Value             | Description                                                                         |
+|:------------------|:------------------------------------------------------------------------------------|
+| `PRESERVE_ALL`    | Preserve the state of all existing dependencies (including recursive dependencies)  |
+| `PRESERVE_DIRECT` | Preserve the state of all existing direct dependencies                              |
+| `PRESERVE_SEMVER` | Preserve semver-compatible versions of direct dependencies                          |
+| `PRESERVE_NONE`   | Do not attempt to preserve any version information                                  |
+| `PRESERVE_TIERED` | Use the tier which will preserve the most version information (this is the default) |
```

Comment:
Worth clarifying here that we try them in order? Or maybe write that in the text block above, I think we could explain the default there too.